agent_app_sample_code/validators/validate_data_pipeline_config.py

```diff
@@ -0,0 +1,223 @@
+# Databricks notebook source
+# Install these if running this Notebook on its own
+# %pip install -U -qqqq databricks-sdk mlflow mlflow-skinny
+# dbutils.library.restartPython()
+
+# COMMAND ----------
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.vectorsearch import EndpointStatusState, EndpointType
+from mlflow.utils import databricks_utils as du
+from databricks.sdk.service.serving import EndpointCoreConfigInput, EndpointStateReady
+from databricks.sdk.errors import ResourceDoesNotExist
+import os
+from databricks.sdk.service.compute import DataSecurityMode
+from pyspark.sql import SparkSession
+
+w = WorkspaceClient()
+browser_url = du.get_browser_hostname()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Check if running on Single User 14.3+ cluster
+
+# COMMAND ----------
+
+
+# Get the cluster ID
+spark_session = SparkSession.getActiveSession()
+cluster_id = spark_session.conf.get("spark.databricks.clusterUsageTags.clusterId", None)
+
+# # Get the current cluster name
+# try:
+#   cluster_id = dbutils.notebook.entry_point.getDbutils().notebook().getContext().tags().get("clusterId").get()
+# except Exception as e:
+
+cluster_info = w.clusters.get(cluster_id)
+
+# Check if a single user cluster
+# Serverless will return None here
+# if not cluster_info.data_security_mode == DataSecurityMode.SINGLE_USER:
+#   raise ValueError(f"FAIL: Current cluster is not a Single User cluster.  This notebooks currently require a single user cluster.  Please create a single user cluster: https://docs.databricks.com/en/compute/configure.html#single-node-or-multi-node-compute")
+
+# Check for 14.3+
+major_version = int(cluster_info.spark_version.split(".")[0])
+minor_version = int(cluster_info.spark_version.split(".")[1])
+
+if not ((major_version==15) or (major_version==14 and minor_version>=3)):
+  raise ValueError(f"FAIL: Current cluster version {major_version}.{minor_version} is less than DBR or MLR 14.3.  Please create a DBR 14.3+ single user cluster.")
+else:
+  print("PASS: Running on a single user cluster version with DBR or MLR 14.3+ ({major_version}.{minor_version}).")
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Check if configured locations exist
+# MAGIC
+# MAGIC If not, creates:
+# MAGIC - UC Catalog & Schema
+# MAGIC - Vector Search Endpoint
+# MAGIC - Source Path
+
+# COMMAND ----------
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors import NotFound, PermissionDenied
+w = WorkspaceClient()
+
+# Create UC Catalog if it does not exist, otherwise, raise an exception
+try:
+    _ = w.catalogs.get(UC_CATALOG)
+    print(f"PASS: UC catalog `{UC_CATALOG}` exists")
```

Comment:
We should probably clean up some of these success `print`s - on successes we don't need to print, and on failures we should just throw an exception

Reply:
i think the prints are helpful bc some of these steps can take some time. otherwise it looks like the code is hanging